Feature/gizfe 676

jj13jp opened the pull request:
## チケットのリンク
https://gizumo.backlog.com/view/GIZFE-676

## やったこと

## やらないこと
- PostList.vueの変更
　→Vuexのstate内部を変更したため、検出されています(selectedCategory→targetCategory)

## テスト
https://gizumo.backlog.com/view/GIZFE-678

## 特にレビューをお願いしたい箇所
- 変数とCSSクラスの命名について
　→別ファイル（article系）を参考に命名したが、もっと良い命名があるのではないかという気がしてならないです
　→一般的にはこう命名するべき、という意見を聞ければ嬉しいです

jj13jp committed:
feat: 動的ルーティング追加(rebased)

categories/:id

test: rebaseするぞ

jj13jp committed:
feat: カテゴリー更新画面のコーディング完成（仮/エラー出ます）

ピクセルパーフェクト狙わないならこれでよし

jj13jp committed:
feat: カテゴリー名の動的表示を実装（仮）

リファクタリングこれから

jj13jp committed:
feat: inputのvalueをリアクティブ化

jj13jp committed:
feat: 更新処理完成（次はメッセージの表示処理）

リファクタリング必須
→メソッド名とdispatchで呼ぶときの命名が一致しておらず混乱を招くため

jj13jp committed:
feat: メッセージ表示実装（成功時/失敗時のそれぞれで）

クラス名（cssの記述）は見直すべきかも？

jj13jp committed:
revert: router.jsをロールバック&ルーティング設定の記述

prettierでフォーマットしてしまっていたのを戻した

jj13jp committed:
refactor: detail→detail"s"にした

jj13jp committed:
refactor: 命名変更（article系のファイルを参考に）

ex )
　・selectedCategory→targetCategory（categories.js）
　・:value→:target-category-name（Edit.vue）

refactor: rebase必須

jj13jp committed:
refactor: Propsの渡し方を修正（仕様書通りでなかった）

:target-category-name→:category

kuyasawa commented:
機能自体は問題なさそうです。
2点気になるところがありました。

![スクリーンショット 2023-03-09 16 00 03](https://user-images.githubusercontent.com/108847433/223945065-5f23fc6a-cc8b-4f32-ae3d-f07be23a734d.png)
エラーがひとつ出てしまっています。

![スクリーンショット 2023-03-09 15 37 52](https://user-images.githubusercontent.com/108847433/223944388-589c9187-4356-48bb-a607-2ca69eedb3d2.png)
エラーの後に更新、また更新の後にエラーになった場合、前のメッセージがそのまま残っています。

API通信中にボタンを押せなくするloadingの機能がついてませんが、これはもしかしたら課題外の機能かもなので、メンターさんに確認お願いします。

jj13jp committed:
fix: Propsの型チェックエラー修正

デフォルトの値を空文字にすることで対処

jj13jp committed:
fix: メッセージ表示処理の修正

更新ボタン押下時、メッセージを必ず消すように変更

jj13jp committed:
fixEX: スペルミス修正（恐らくマスターブランチ）

ryo-kihara commented on src/components/pages/Categories/Edit.vue:
```diff
@@ -0,0 +1,51 @@
+    },
+  },
+  created() {
+    this.$store.dispatch('categories/getCategoryDetails', this.$route);
```
[imo]

`this.$route` を引数として渡すと上記の記述をパッと見たときに何をしているのかがわかりづらいと個人的に思います。`this.$route`がどんなふうに使われているかがわかりにくいからです。

以下のような記述なら「カテゴリの詳細を id から取得しているんだな」というのがぱっと見でわかりやすいと思います。

```suggestion
    this.$store.dispatch('categories/getCategoryDetails', this.$route.params.id);
```

jj13jp replied:
修正しました！

ryo-kihara commented on src/components/pages/Categories/Edit.vue:
```diff
@@ -0,0 +1,51 @@
+  created() {
+    this.$store.dispatch('categories/getCategoryDetails', this.$route);
+  },
+  destroyed() {
+    this.$store.dispatch('categories/inputTargetCategoryName', '');
+  },
```
👍

jj13jp committed:
feat: ボタンにloading機能を追加

ボタンはAPI通信中、
・disabledかつcursor: auto状態
・連打を拒否（if (state.isLoading === true) return;）

ryo-kihara commented on src/components/pages/Categories/Edit.vue:
```diff
@@ -0,0 +1,51 @@
+      :error-message="errorMessage"
+      :done-message="doneMessage"
+      @update-value="updateValue"
+      @click="updateCategoryName"
```
[imo]

こちらもぱっと見になりますが、`@click` だと何をクリックしたのかがわかりづらく、子コンポーネントの実装を見にいく必要が出てきてしまうかなと思います。

`@handle-submit` などであればどんなときに発火されるイベントなのかがわかりやすくなるかなと思います。

命名とかはめちゃ大切です。

jj13jp replied:
修正しました。

ryo-kihara commented:
> 変数とCSSクラスの命名について
> 　→別ファイル（article系）を参考に命名したが、もっと良い命名があるのではないかという気がしてならないです
> 　→一般的にはこう命名するべき、という意見を聞ければ嬉しいです

主に関数のところでコメントしました！
変数と CSS は個人的には気にならないかなと思ったのですが、逆にどのあたりが気になったとかはありますか？

jj13jp committed:
refactor: 可読性向上fix他

・@click→@handle-submit
・this.$route→this.$route.params.id
・updateCategoryName→updateTargetCategoryName

Joex13 commented:
> > API通信中にボタンを押せなくするloadingの機能がついてませんが、これはもしかしたら課題外の機能かもなので、メンターさんに確認お願いします。
> 
> こちら対応お願いします！

対応しました！

ryo-kihara commented on src/components/pages/Categories/Edit.vue:
```diff
@@ -0,0 +1,55 @@
+    this.$store.dispatch('categories/inputTargetCategoryName', '');
+  },
+  methods: {
+    updateValue(event) {
+      this.$store.dispatch(
+        'categories/inputTargetCategoryName',
+        event.target.value,
+      );
+    },
+    updateTargetCategoryName() {
+      this.$store.dispatch('categories/updateTargetCategoryName');
+    },
```
> "updateValue"と"inputTargetCategoryName"で、
> 記述に相違があることに違和感を感じています。

なるほど！
確かにここに関しては気になるかもしれないです。

以下のような記述が一番わかりやすいかなと思うのですが、いかがでしょう。

```suggestion
    updateTargetCategoryName(event) {
      this.$store.dispatch(
        'categories/updateTargetCategoryName',
        event.target.value,
      );
    },
    handleSubmit() {
      this.$store.dispatch('categories/updateCategory');
    },
```

- updateTargetCategoryName(event)
  - 入力中の文字で更新対象のカテゴリ名を更新していく
- handleSubmit()
  - 更新が確定して送信する
- this.$store.dispatch('categories/updateCategory')
  - 〇〇Category（〇〇は CRUD などの動詞）で API を実行する actions というように統一する。

jj13jp committed:
refactor: 可読性向上（メソッド名の変更）

自己満かもしれない

jj13jp committed:
refactor: updateTargetCategoryName→updateCategory
